hazelcast-sql/src/test/java/com/hazelcast/sql/SqlErrorAbstractTest.java

```diff
@@ -309,4 +316,44 @@ protected HazelcastInstance newClient() {
     protected ClientConfig clientConfig() {
         return new ClientConfig();
     }
+
+    protected static void assertErrorCode(int expected, HazelcastSqlException error) {
+        assertEquals(error.getCode() + ": " + error.getMessage(), expected, error.getCode());
+    }
+
+    /**
+     * Start the new Hazelcast instance.
+     *
+     * @param awaitAssignment whether to wait for a partition assignment to a new member
+     * @return created instance
+     */
+    protected HazelcastInstance newHazelcastInstance(boolean awaitAssignment) {
```

Comment:
Can we rely on the family of `waitAllForSafeState` methods?

Reply:
This method cannot guarantee that every node has at least one partition. Therefore, `testMapMigration` could fail again.